src/views/Entries/index.js

```diff
@@ -145,20 +149,163 @@ const leadKeySelector = d => d.id;
 
 const MAX_ENTRIES_PER_REQUEST = 50;
 
+const requestOptions = {
+    geoOptionsRequest: {
+        url: '/geo-options/',
+        query: ({ props }) => ({
+            project: props.projectId,
+        }),
+        onMount: true,
+        onPropsChanged: ['projectId'],
+        onSuccess: ({ response, props, params }) => {
+            const {
+                setGeoOptions,
+                projectId,
+            } = props;
+
+            const {
+                onGeoOptionsSuccess,
+            } = params;
+
+            setGeoOptions({
+                projectId,
+                locations: response,
+            });
+
+            onGeoOptionsSuccess();
+        },
+        onFailure: ({ response }) => {
+            const message = response.$internal.join(' ');
+            notify.send({
+                title: _ts('entries', 'entriesTabLabel'),
+                type: notify.type.ERROR,
+                message,
+                duration: notify.duration.MEDIUM,
+            });
+        },
+        onFatal: () => {
+            notify.send({
+                title: _ts('entries', 'entriesTabLabel'),
+                type: notify.type.ERROR,
+                message: _ts('entries', 'geoOptionsFatalMessage'),
+                duration: notify.duration.MEDIUM,
+            });
+        },
+        extras: {
+            schemaName: 'geoOptions',
+        },
+    },
+    projectFrameworkRequest: {
+        url: ({ props: { projectId } }) => `/projects/${projectId}/analysis-framework/`,
+        onMount: true,
+        onPropsChanged: ['projectId'],
+        onSuccess: ({ response, props, params }) => {
+            const { setFramework } = props;
+            const { onFrameworkGetSuccess } = params;
+
+            setFramework({
+                analysisFramework: response,
+            });
+
+            onFrameworkGetSuccess();
+        },
+        extras: {
+            schemaName: 'analysisFramework',
+        },
+    },
+    entriesRequest: {
+        url: '/entries/filter/',
+        query: ({ props }) => ({
+            offset: (props.activePage - 1) * MAX_ENTRIES_PER_REQUEST,
+            limit: MAX_ENTRIES_PER_REQUEST,
+        }),
+        method: methods.POST,
+        body: ({ props }) => {
+            const {
+                geoOptions,
+                framework,
+                projectId,
+                entriesFilters: widgetFilters,
+            } = props;
+
+            const otherFilters = {
+                project: projectId,
+            };
+
+            const processedFilters = processEntryFilters(
+                widgetFilters,
+                framework,
+                geoOptions,
+            );
+
+            return ({
+                filters: [
+                    ...processedFilters,
+                    ...Object.entries(otherFilters),
+                ],
+            });
+        },
+        onMount: false,
+        onPropsChanged: ['entriesFilter', 'activePage'],
```

Comment:
Shouldn't this also be called on onMount?

Reply:
No, we only fetch entries after geooptions and frameworks are loaded to avoid unintended data references.

Reply:
There are various limitations with this apporach when we have entriesFilter/activePage as triggers.
And, the data loading is also slow.
We should rather focus on fixing the bad-data reference (we can just hide entries unless geo-options and frameworks are loaded)